vendor/patch/src/patch.ml

```diff
@@ -52,18 +52,31 @@ type hunk = {
 }
 
 let unified_diff hunk =
+  let left_line_num = ref hunk.their_start in 
+  let right_line_num = ref hunk.mine_start in
```

Comment:
I think these two are swapped. `right_line_num` is initialized with `hunk.mine_start` but incremented on each `` `Their ``.

I suggest using "mine" and "their" in the variable name to avoid confusion.